TypeScript: Types for createEditSession do not allow for string modes

### Describe the bug

The types for `createEditSession` say that the optional second argument, `mode`, can only be `Ace.SyntaxMode`:

https://github.com/ajaxorg/ace/blob/c89c0d56d862a9db39161eb870945b29a3ea6c95/ace.d.ts#L1062

However, the underlying constructor accepts `Ace.SyntaxMode` or a `string`:

https://github.com/ajaxorg/ace/blob/c89c0d56d862a9db39161eb870945b29a3ea6c95/src/edit_session.js#L40



### Expected Behavior

The ability to pass a string like `'ace/mode/rust'` as the second argument without TypeScript complaining.

### Current Behavior

TypeScript complains:

```
index.ts:3:27 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'SyntaxMode'.

3 ace.createEditSession('', 'ace/modes/rust');
                            ~~~~~~~~~~~~~~~~
```

### Reproduction Steps

```
mkdir repro
cd repro
pnpm init
pnpm add ace-builds typescript
pnpm tsc --init
cat >index.ts <<EOF
import * as ace from 'ace-builds';

ace.createEditSession('', 'ace/modes/rust');
EOF
pnpm tsc
```

### Possible Solution

Enhance the types to match.

### Additional Information/Context

_No response_

### Ace Version / Browser / OS / Keyboard layout

ace-builds 1.43.6; typescript 6.0.3

--- a/ace.d.ts
+++ b/ace.d.ts
@@ -1059,7 +1059,7 @@ declare module "ace-code" {
         env?: any;
         value?: any;
     }) | null, options?: Partial<import("ace-code").Ace.EditorOptions>): Editor;
-    export function createEditSession(text: import("ace-code/src/document").Document | string, mode?: import("ace-code").Ace.SyntaxMode): EditSession;
+    export function createEditSession(text: import("ace-code/src/document").Document | string, mode?: import("ace-code").Ace.SyntaxMode | string): EditSession;
     import { Editor } from "ace-code/src/editor";
     import { EditSession } from "ace-code/src/edit_session";
     import { Range } from "ace-code/src/range";

--- a/demo/test_package/index.ts
+++ b/demo/test_package/index.ts
@@ -34,6 +34,7 @@ const editor = ace.edit(null, {
 }); // should not be an error
 editor.setTheme("ace/theme/monokai");
 editor.session.setMode("ace/mode/javascript");
+ace.createEditSession("", "ace/mode/rust"); // should not be an error
 
 function configure(config: ace.Ace.Config) {
     config.setDefaultValues("editor", {

--- a/src/ace.js
+++ b/src/ace.js
@@ -73,7 +73,7 @@ exports.edit = function(el, options) {
 /**
  * Creates a new [[EditSession]], and returns the associated [[Document]].
  * @param {import('./document').Document | String} text {:textParam}
- * @param {import("../ace-internal").Ace.SyntaxMode} [mode] {:modeParam}
+ * @param {import("../ace-internal").Ace.SyntaxMode | string} [mode] {:modeParam}
  * @returns {EditSession}
  **/
 exports.createEditSession = function(text, mode) {

